frontend/src/app/blog/[name]/[id]/page.tsx

```diff
@@ -0,0 +1,125 @@
+"use client";
+import Image from "next/image";
+import { useParams } from "next/navigation";
+import BlogsDataJson from "../../../../data/blogs.json";
+import { BlogsDataType } from "@/types";
+
+const Blog = () => {
+  const BlogsData: BlogsDataType = BlogsDataJson;
+  const params = useParams();
+  const blogId = params.id as string;
+  const blogData = BlogsData.entries[blogId];
+
+  if (!BlogsDataJson.ids.find((id) => id === blogId)) {
+    return (
+      <div className="flex-1 grid place-content-center">
+        <h1 className="flex items-center gap-2">
+          <span className="text-3xl">404 |</span>
+          <span>This page could not be found.</span>
+        </h1>
+      </div>
+    );
+  }
+
+  return (
+    <section className="p-2">
```

Comment:
You are not using the theme from the theme provider so the page is perpetually in dark mode. please fix it

Reply:
done